plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,363 @@
+		resourceToServicesMap[resource] = append(resourceToServicesMap[resource], service)
+	}
+
+	var resources []transit.DynamicMonitoredResource
+	for resourceName, services := range resourceToServicesMap {
+		resources = append(resources, transit.DynamicMonitoredResource{
+			BaseResource: transit.BaseResource{
+				BaseTransitData: transit.BaseTransitData{
+					Name: resourceName,
+					Type: transit.Host,
+				},
+			},
+			Status:        transit.HostUp,
+			LastCheckTime: transit.NewTimestamp(),
+			Services:      services,
+		})
+	}
+
+	traceToken, _ := uuid.GenerateUUID()
+	requestJSON, err := json.Marshal(transit.DynamicResourcesWithServicesRequest{
+		Context: &transit.TracerContext{
+			AppType:    "TELEGRAF",
+			AgentID:    g.AgentID,
+			TraceToken: traceToken,
+			TimeStamp:  transit.NewTimestamp(),
+			Version:    transit.ModelVersion,
+		},
+		Resources: resources,
+		Groups:    nil,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	headers := map[string]string{
+		"GWOS-APP-NAME":  "telegraf",
+		"GWOS-API-TOKEN": g.authToken,
+		"Content-Type":   "application/json",
+		"Accept":         "application/json",
+		"User-Agent":     internal.ProductToken(),
+	}
+
+	statusCode, body, err := clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+	if err != nil {
+		return fmt.Errorf("error while sending: %v", err)
+	}
+
+	/* Re-login mechanism */
+	if statusCode == 401 {
+		if err = g.Connect(); err != nil {
+			return fmt.Errorf("re-login failed: %v", err)
+		}
+		headers["GWOS-API-TOKEN"] = g.authToken
+		statusCode, body, err = clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+		if err != nil {
+			return fmt.Errorf("error while sending: %v", err)
+		}
+	}
+
+	if statusCode != 200 {
+		return fmt.Errorf("HTTP request failed. [Status code]: %d, [Response]: %s", statusCode, string(body))
+	}
+
+	return nil
+}
+
+func (g *Groundwork) Description() string {
+	return "Send telegraf metrics to GroundWork Monitor"
+}
+
+func init() {
+	outputs.Add("groundwork", func() telegraf.Output {
+		return &Groundwork{
+			ResourceTag:         "host",
+			DefaultHost:         "telegraf",
+			DefaultServiceState: string(transit.ServiceOk),
+		}
+	})
+}
+
+func (g *Groundwork) parseMetric(metric telegraf.Metric) (string, transit.DynamicMonitoredService) {
+	resource := g.DefaultHost
+	if value, present := metric.GetTag(g.ResourceTag); present {
+		resource = value
+	}
+
+	service := metric.Name()
+	if value, present := metric.GetTag("service"); present {
+		service = value
+	}
+
+	status := g.DefaultServiceState
+	value, statusPresent := metric.GetTag("status")
+	if validStatus(value) {
+		status = value
+	}
+
+	message := ""
+	if value, present := metric.GetTag("message"); present {
+		message = value
+	}
+
+	unitType := string(transit.UnitCounter)
+	if value, present := metric.GetTag("unitType"); present {
+		unitType = value
+	}
+
+	critical := -1.0
```

Comment:
```suggestion
	var critical float64
```

Reply:
Why? With your change, the default value (if the tag does not contain a float or is not present) will be `0.0`. I guess it is ` -1.0` to guarantee filtering below is always false/true...

Reply:
It causes problems at the receiving endpoint. The critical value should only be used in case of when it has actual been set.

Currently, if you have a critical value that isn't parsable to float, you will end up with a metric with -1.0 as threshold.

Reply:
"-1" - is a default value for all thresholds. This is not a problem for receiving endpoint to process it

Reply:
Maybe its not a problem at the endpoint (I highly doubt this, since how to specify a threshold at -1 then?) but it introduces some unexpected behaviour in the used `transit.CalculateServiceStatus` method.

Reply:
Fixed